Runtime/Scripts/NoCodes/Internal/NoCodesInternal.cs

```diff
@@ -0,0 +1,301 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using QonversionUnity.MiniJSON;
+using UnityEngine;
+
+namespace QonversionUnity
+{
+    /// <summary>
+    /// Internal implementation of NoCodes SDK.
+    /// </summary>
+    internal class NoCodesInternal : MonoBehaviour, INoCodes
+    {
+        private const string GameObjectName = "NoCodesRuntimeGameObject";
+        private const string SdkVersion = "10.0.3";
+        private const string SdkSource = "unity";
+
+        // Event method names for Unity callbacks
+        private const string OnScreenShownMethodName = "OnNoCodesScreenShown";
+        private const string OnFinishedMethodName = "OnNoCodesFinished";
+        private const string OnActionStartedMethodName = "OnNoCodesActionStarted";
+        private const string OnActionFailedMethodName = "OnNoCodesActionFailed";
+        private const string OnActionFinishedMethodName = "OnNoCodesActionFinished";
+        private const string OnScreenFailedToLoadMethodName = "OnNoCodesScreenFailedToLoad";
+
+        private INoCodesWrapper _nativeWrapperInstance;
+
+        // Event handler
+        private Action<NoCodesEvent> _onEventReceived;
+
+        // Delegate for handling NoCodes events
+        [CanBeNull]
+        private NoCodesDelegate _delegate;
+
+        public event Action<NoCodesEvent> EventReceived
+        {
+            add
+            {
+                _onEventReceived += value;
+            }
+            remove
+            {
+                _onEventReceived -= value;
+            }
+        }
+
+        public static NoCodesInternal CreateInstance()
+        {
+            GameObject go = new GameObject(GameObjectName);
+            NoCodesInternal instance = go.AddComponent<NoCodesInternal>();
+            DontDestroyOnLoad(go);
+
+            return instance;
+        }
+
+        void INoCodes.InitializeInstance(NoCodesConfig config)
+        {
+            INoCodesWrapper instance = GetNativeWrapper();
+            instance.Initialize(GameObjectName);
+            instance.InitializeNoCodes(config.ProjectKey, SdkVersion, SdkSource, config.ProxyUrl);
+        }
+
+        public void SetDelegate(NoCodesDelegate @delegate)
+        {
+            _delegate = @delegate;
+        }
+
+        public Task SetScreenPresentationConfig(NoCodesPresentationConfig config, [CanBeNull] string contextKey = null)
+        {
+            try
+            {
+                var configDict = config.ToDictionary();
+                string configJson = Json.Serialize(configDict);
```

Comment:
In the `ScreenPresentationConfig` class (which is not removed afais, but should be) there was a `ToJson` method, encapsulating this logic. Let's do the same here?